### ISSUE TITLE
Possibility for an illustrated guide for disabled people

### ISSUE THREAD

**Issue by Rumastiker** [nextcloud/cookbook#324]
**Is your feature request related to a problem? Please describe.**
I can't add pictures to the steps. I need it for easy language recipes.

**Describe the solution you'd like**
Inserting pictures in the text boxes

**Describe alternatives you've considered**
Because only one picture is stored, I see no alternative.

Excuse my simple English sentences, my mother tongue is German. I had to use the google translator ;-)

**Comment by christianlupus** [comment]
Hello. I will answer in German to simplify the communication here. Will try to summarize if possible later.

@Rumastiker Bei den Textfeldern können leider keine Bilder eingebunden werden. Diese enthalten ja nur reinen Text im Moment. Es gab zwar Überlegungen, hier auch Markdown zuzulassen, das ist meines Wissens nach aber nicht realisiert worden.

Ziel bei der App ist es zum Standard konform zu bleiben, der auf [Schema.org](https://schema.org/Recipe) festgelegt ist. Dort gäne es theoretisch die Möglichkeit, dass Bilder **zwischen** einzenlen Texten auftauchen. Alternativ könnte man versuchen oben mehrere Bidler in der Kopfzeile zuzulassen.

Würde das helfen?

**Comment by Rumastiker** [comment]
Hallo Christian,
n-Zeilen Text der 'Zubereitung' und abwechsend n-Zeilen Bilder wäre genauso gut und würde das Problem beheben.  Das wäre schon einmal super, wenn es dem Standard entspricht! Mehr Bilder in der Kopfzeile wäre wahrscheinlich nett, aber nicht dringend nötig. Gerade Menschen mit Handicap (wie mein Sohnemann) benötigen einfache Texte, am Besten mit einem Bild des Arbeitsschrittes.

LG Andreas

**Comment by sam-19** [comment]
I think the idea has been to support Markdown in the instructions section. Markdown can achieve this, but you would still have to host the images in a separate location. In the long run, it would be great if recipes could have file attachments, such as images to embed in the text.

**Comment by Rumastiker** [comment]
Hallo Sam.
Now I had to google what Markdown is. Or what Markdown has to do with the problem. If I got it right, Markdown is a simplified 'formatting language'. I understand you want to embed an image in text boxes using the markdown command. Or are all recipes formatted with Markdown? As a user, I don't really care, the main thing is that it works ;-) As a supervisor of end users, I would advise against the manual formatting of characters in the text. Or did you mean that Markdown is used when transferring recipes and that the image must also be transferred when transferring? So, before you argue, clarify the terms.

LG Andreas

**Comment by sam-19** [comment]
@Rumastiker yes, Markdown is a simple text formatting syntax, used by GitHub for example. It was actually @christianlupus who was talking about Markdown and my reply was partly aimed to him. To clarify: using Markdown to display images in the preparation steps would not break recipe schema, because you could provide a written instruction in the image alt tag and when exporting a recipe, just strip the image part leaving the alt text. So *displaying images* does not require any additional features, Markdown support (once it happens) will make that possible. But, as I mentioned above, the images would still have to be hosted somehow, and that would need some kind of a plan. Maybe Christian can give us his two cents on that.

**Comment by christianlupus** [comment]
@sam-19 no, we are in fact not talking of the same thing. The [Recipe](https://schema.org/Recipe) has a property `recipeInstruction` that can be of type [CreativeWork](https://schema.org/CreativeWork). This in itself can be an image (as URL or ImageObject). The idea was to have a mixture of plain text (the real instructions) and interactive content (even videos would be possible by this).

I was considering the following regarding the file management: We have already a folder per recipe. We could add more files/images/.... to that folder and allow to access it interactively. There we could add these additional images once we have a clear structure. The same structure I have in mind with #323, #315 and #135. 

Nevertheless, adding Markdown support might be another issue #47. There the request is markdown itself (although with intention for images as well).
